core/src/gossip/orderbook.rs

```diff
@@ -318,6 +324,15 @@ impl GossipProtocolExecutor {
         let reblind_proof = proof_bundle.copy_reblind_proof();
         let commitment_proof = proof_bundle.copy_commitment_proof();
 
+        // Verify the two proofs are correctly commitment linked
+        if !verify_reblind_commitments_link(&reblind_proof.commitment, &commitment_proof.commitment)
```

Comment:
down the line we should use this for peer scoring / banning

Reply:
Yeah good call out; we store priorities at both the cluster and order granularity in the global state object [here](https://github.com/renegade-fi/renegade/blob/main/core/src/state/priority.rs#L54-L59). I was thinking we just add hooks to update these values every time a rule like the one you're mentioning occurs

Reply:
yeah i like that - definitely doesn't need to be in this PR, though i hope that hooking in to update the global state doesn't introduce too much locking overhead...
it probably won't so this train of thought may be premature optimization, but we could also move all peer priority updates to a separate (low priority) task